script.js

```diff
@@ -1,4 +1,312 @@
+  return `${card.name} of ${convertSuitWordToEmoji(card.suit)}`;
+};
+
+var printCardswithEmojiSuits = function (cards) {
+  var returnString = '';
+  // Iterate until cards.length - 1 so we can avoid the extra comma at the end of return string
+  for (var i = 0; i < cards.length - 1; i += 1) {
+    var currCard = cards[i];
+    returnString += `${getCardStringRepresentation(currCard)}, `;
+  }
+  var lastCard = cards[cards.length - 1];
+  returnString += getCardStringRepresentation(lastCard);
+  return returnString;
+};
+
+var howToPlay = function() {
+  return `To play BlackJack, first place a bet.<br>
+  The aim of Black Jack is to draw cards up to 21<br>
+  After drawing 2 cards. Decide to hit to draw<br>
+  or stand to stop drawing. Cards are compared<br>
+  with computer's hand. The hand that is closest to 21 wins.<br><br>
+
+  Enjoy the Game!`
+}
+
+// features to work on
+//1. reset blackjack game game ends (solved)
+//2. change suit to emoji
+//3. collect user name and display it
+//4. betting function
+
+// initalise global array to store the cards
+
+var playerCards = [];
+var computerCards = [];
+var gameMode = `username`;
+var playerScore = 0;
+var computerScore = 0;
+var winMessage= ``;
+var playerPoints = 100;
+var userName = ``;
+var betPoints
+
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+
+  if (!userName) {
+    if (!input) {
+      return 'Please input a non-empty string as your user name!';
+    }
+    userName = input;
+    gameMode = `betpoints`
+    return winMessage = `Let's welcome ${userName} to our BlackJack game!<br><br>
+    You have ${playerPoints} points.<br><br>
+    How many points do you want to bet?`
+  }
+
+  if (gameMode == `reset`){
+    playerCards = [];
+    computerCards = [];
+    playerScore = 0;
+    computerScore = 0;
+    gameMode = `betpoints`
+    return winMessage = `${userName}, Let play another game.<br><br>
+    You have ${playerPoints} points.<br><br>
+    How many points do you want to bet?`
+  }
+
+  if (gameMode == `betpoints`) {
+    gameMode = `draw card`
+    betPoints = Number(input)
+    return winMessage = `You have bet ${betPoints} Points.<br><br>
+    Press Submit to Deal Cards.`
+  }
+
+  if (gameMode === `draw card`) {
+    for (i = 0; i < 2; i++) {
+      playerCards.push(shuffledDeck.pop());
+    };
+    //summarise score and assign to global variable
+    playerScore = getHandSum(playerCards);
+    gameMode = `hit or stand`;
+  }
+
+  winMessage= `You drawed ${printCardswithEmojiSuits(playerCards)}
+  <br><br>
+  Your score is: ${playerScore}.<br><br>
+  Enter hit or stand.`
+
+  if (gameMode == `hit or stand`) {
+
+    if (input == `hit`){
+
+      playerCards.push(shuffledDeck.pop());
+      playerScore = getHandSum(playerCards);
+      console.log(playerCards)
+      var drawnCard = `${userName}, now these cards are in your hand `
```

Comment:
Good use of template literals (i.e. ``${}`` . Keeps your code concise compared to the usual String concatenation